components/raftstore-v2/src/operation/command/mod.rs

```diff
@@ -148,6 +148,7 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
             .apply_pool
             .spawn(async move { apply_fsm.handle_all_tasks().await })
             .unwrap();
+        fail::fail_point!("deplay_set_apply_scheduler", |_| {});
         self.set_apply_scheduler(apply_scheduler);
```

Comment:
Why not set it before spawning?

Reply:
Is there any benefit to put it before spawning? The spawning itself is not going to take long time either.
The key here is to delay the set_apply_scheduler being called, so making this fail_point just prior to it is more precise. Though, in fact the delay could happen at any place when the OS thread is switched.

Reply:
> Is there any benefit to put it before spawning? The spawning itself is not going to take long time either.
> The key here is to delay the set_apply_scheduler being called, so making this fail_point just prior to it is more precise. Though, in fact the delay could happen at any place when the OS thread is switched.

Is there any benefit to put it before spawning? The spawning itself is not going to take long time either.
The key here is to delay the set_apply_scheduler being called, so making this fail_point just prior to it is more precise. Though, in fact the delay could happen at any place when the OS thread is switched.

Reply:
> Why not set it before spawning?

Is there any benefit to put it before spawning? The spawning itself is not going to take long time either.
The key here is to delay the set_apply_scheduler being called, so making this fail_point just prior to it is more precise. Though, in fact the delay could happen at any place when the OS thread is switched.

Reply:
Forget it, I thought it's a race between `Apply::handle_all_tasks`.